### PR TITLE
[frames] Helper functions for frames

### PR DESCRIPTION
**Pull request by gabrielebndn**
framesForwardKinematics(model, data, frame_id) computes oMf[frame_id], stores it and returns it. Assumes forwardKinematics has already been run. Thought it might be useful if the model contains many frames but we are only interested in one or two of them.

getFrameVelocity(model, data, frame_id, v_frame) computes the velocity of frame frame_id and puts it in v_frame. Assumes first-order forwardKinematics has already been run.

getFrameAcceleration(model, data, frame_id, a_frame) computes the acceleration of frame frame_id and puts it in a_frame. Assumes second-order forwardKinematics has already been run.

### CODE REVIEW

**Comment by proyan** [comment]
I think you should also implement the python bindings, and update the robot_wrapper API which does the same thing.

**Comment by gabrielebndn** [comment]
I agree. I was planning to do it if this pull request gets accepted

File: src/algorithm/frames.hpp [hunk 1]
```diff
@@ -48,22 +48,70 @@ namespace se3
                                       const Eigen::VectorXd & q);
 
   /**
-   * @brief      Returns the jacobian of the frame expresssed either expressed in the LOCAL frame coordinate system or in the WORLD coordinate system,
+   * @brief      Updates the position of the given frame.
```
Comment (jcarpent) [inline src/algorithm/frames.hpp]: placement instead position

**Comment by jcarpent** [comment]
Thanks for these feedbacks.

I would go for updateFramePlacement instead of framesForwardKinematics by making this last function deprecated.

**Comment by jcarpent** [comment]
@gabrielebndn Can you make the modifications?

**Comment by gabrielebndn** [comment]
Alright

**Comment by gabrielebndn** [comment]
Ok, to summarize:
- framesForwardKinematics(model,data,q) stays the same
- framesForwardKinematics(model,data) is deprecated and replaced by updateFramePlacements
- updateFramePlacements(model,data) replaces framesForwardKinematics(model,data)
- updateFramePlacement(model,data,frame_id) [without final s] updates and returns a single frame
- getFrameVelocity(model, data, frame_id, v_frame) returns the velocity of a single frame
- getFrameAcceleration(model, data, frame_id, v_frame) returns the acceleration of a single frame

So, are we sure we want to split framesForwardKinematics into two different methods depending on the arguments? It might be confusing for those who are already using these methods. Maybe it's just easier to stay with framesForwardKinematics.

In Python, I also did the following
- framesKinematics(model,data[,q]) is deprecated. It should be replaced either by framesForwardKinematics(model,data,q) or updateFramePlacements(model,data). I deprecated this function because the name was not consistent with C++
- created the bindings for the new methods
- updated RobotWrapper to use the bindings
- other changes to RobotWrapper. Notably, deprecated "position" and "framePosition" to replace them with "placement" and "framePlacement"

**Comment by gabrielebndn** [comment]
Update: I slightly modified deprecation.py so that the instructions are automatically used as docstring in case none is provided

**Comment by jcarpent** [comment]
Looks fine.
I'm going to merge!

**Pull request merged**